### PR TITLE
rc: Changed parsing to use Pyparsing library to get a better aproach.

### PR DESCRIPTION
**Pull request by nijel**
I've rebased and squashed changes from https://github.com/translate/translate/compare/master...PhoneixS:pyparsing-rc created originally by @PhoneixS

See https://github.com/translate/translate/issues/3204

What was omitted:

* https://github.com/translate/translate/commit/284324221f539f725d6b0e9e797752c997584f2e seems unrelated and in case it is needed should go to separate PR
* https://github.com/translate/translate/commit/d0b988f1164b902db9717e7e90283c18fcb96a1c changes many lines and seems to do no functional changes

### CODE REVIEW

**Comment by nijel** [comment]
Ready for review, this fixed nearly all reported RC related issues.

File: translate/convert/po2rc.py [hunk 1]
```diff
@@ -16,47 +16,280 @@
...
+import codecs
+from collections.abc import Iterable
+from translate.storage.rc import generate_popup_caption_name,\
+    generate_popup_pre_name, generate_menu_pre_name
+
 
 from translate.convert import convert
 from translate.storage import po, rc
 
+NL = "\n"
```
Comment (nschonni) [inline translate/convert/po2rc.py:34]: Should this be `\r\n` if it's a windows specific format?

Comment (nijel) [reply translate/convert/po2rc.py:34]: Probably yes. It's coming from https://github.com/translate/translate/commit/86e90ea3760e5b00f19c37e022d399de87262342 and I really didn't touch the logic there...

File: translate/storage/test_rc.py [hunk 2]
```diff
@@ -299,3 +299,81 @@ def test_parse_no_language(self):
...
+CAPTION "Select Media Type"
+FONT 8, "MS Shell Dlg", 400, 0, 0x1
+BEGIN
+    COMBOBOX        IDC_COMBO1,5,5,215,37,CBS_DROPDOWN | WS_VSCROLL | WS_TABSTOP
+    DEFPUSHBUTTON   "OK",IDOK,116,28,50,14
+    PUSHBUTTON      "Cancel",IDCANCEL,170,28,50,14
+END
+"""
+        rc_file = self.source_parse(rc_source)
+        assert len(rc_file.units) == 3
```
Comment (unho) [inline translate/storage/test_rc.py:379]: Maybe we should explicitly check for the content of each of those units as well.

File: translate/storage/test_rc.py [hunk 3]
```diff
@@ -288,3 +288,14 @@ def test_parse_newlines_cr(self):
...
+    def test_parse_no_language(self):
+        """Test parsing a RC string with missing language tag."""
+        rc_source = """
+STRINGTABLE
+BEGIN
+    IDP_REGISTRONOV         "Data isn't valid"
+END
+"""
+        rc_file = self.source_parse(rc_source)
+        assert len(rc_file.units) == 1
```
Comment (unho) [inline translate/storage/test_rc.py:301]: Also check unit content.

File: translate/convert/test_rc2po.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        options = super().test_help(capsys)
+        options = self.help_check(options, "-t TEMPLATE, --template=TEMPLATE")
+        options = self.help_check(options, "-l LANG, --lang=LANG")
+
+    def test_convert(self):
+        """ Tests the conversion to a po file """
+        self.create_testfile("simple.rc", RC_SOURCE)
+        self.run_command(i="simple.rc", o="simple.po")
+        po_result = pofile(self.open_testfile("simple.po"))
+        assert len(po_result.units) == 15
```
Comment (unho) [inline translate/convert/test_rc2po.py:67]: Also check for some of those units, or even for all of them. Same in `TestPO2RCCommand` above.

**Comment by unho** [comment]
@nijel I also suggest adding more tests to increase coverage. Codecov is useful to identify such paths.

**Comment by nijel** [comment]
I've extended the coverage a bit more.

**Pull request merged**